soc/arm/st_stm32/stm32wba/power.c

```diff
@@ -106,10 +171,23 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 			LOG_DBG("Unsupported power substate-id %u",
 							substate_id);
 		}
-	case PM_STATE_STANDBY:
-		/* To be tested */
-		LL_LPM_EnableSleep();
+		break;
 	case PM_STATE_SUSPEND_TO_RAM:
+#if defined(CONFIG_PM_S2RAM)
+		/* Update CMSIS SystemCoreClock variable (HCLK) */
+		/* At reset, system core clock is set to 16 MHz from HSI */
+		SystemCoreClock = 16000000;
+
+		LL_AHB4_GRP1_EnableClock(LL_AHB4_GRP1_PERIPH_PWR);
+
+		LL_LPM_DisableSleepOnExit();
+		LL_LPM_EnableSleep();
```

Comment:
I guess this is the same content as `stm32wba_init()`. It would be preferable to use it directly in case some additions  are made in this function.